packages/compiler-sfc/src/compileScript.ts

```diff
@@ -692,11 +692,9 @@ export function compileScript(
           )}, required: ${required}${
             defaultString ? `, ${defaultString}` : ``
           } }`
-        } else if (type.indexOf('Boolean') > -1) {
+        } else if (type.some(el => el === 'Boolean' || el === 'Function')) {
```

Comment:
Optimization: when type is `Function`, prod type is only needed if `default` is also used.